server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/JdbcBasePlugin.java

```
@@ -285,15 +285,19 @@ private static void closeConnection(Connection connection) throws SQLException {
         }
         try {
             if (
-                !connection.isClosed() &&
-                connection.getMetaData().supportsTransactions() &&
-                !connection.getAutoCommit()
+                    !connection.isClosed() &&
+                            connection.getMetaData().supportsTransactions() &&
+                            !connection.getAutoCommit()
             ) {
                 connection.commit();
             }
-        }
-        finally {
-            connection.close();
+        } finally {
+            try {
+                connection.close();
+            } catch (Exception e) {
+                // ignore
+                LOG.warn("Failed to close JDBC connection, ignoring the error.");

```

Why warn instead of error ?

---

previously, if the operation is done and committed successfully, but there was an error during `connection.close()`, that error would result in an exception and aborting of GPDB transaction. We changed it here so that a failure to close a connection will be just a warning (we ignore the Exception).

@leskin-in -- any concerns with this change of behavior ? 
@oliverralbertini -- we should probably log an exception message if not the whole stack trace.

---

As far as I understand, error on `close()` would not lead to an exception being propagated to GPDB.  [This code](https://github.com/greenplum-db/pxf/blob/148676a7f6441bf10c0b6e3e7aae1e6ae43ef04f/server/pxf-service/src/main/java/org/greenplum/pxf/service/bridge/ReadBridge.java#L139-L146) logs the exception (and I suppose, if we want to change log level, we should do it there), and [this](https://github.com/greenplum-db/pxf/blob/148676a7f6441bf10c0b6e3e7aae1e6ae43ef04f/server/pxf-service/src/main/java/org/greenplum/pxf/service/rest/BridgeResource.java#L152-L156) prevents exceptions from propagation to GDPB.

That said, I think we should either left this unchanged or modify the log level in `BridgeResource`.

---

@leskin-in thanks for the input. In this case the concern is not so much that an exception bubbles out, but that an `SQLException` that we care about gets masked by an exception from closing the connection.

@denalex @shivzone I am switching to logging the entire stack trace here.